### PR TITLE
Add `aria-haspopup` and `aria-expanded` to invoker

### PR DESCRIPTION
**Pull request by JoyceZhu**
This provides screen reader users more information about the state and behavior of a `SelectPanel`'s invoker button.

Going to request some Lookbook testing as well for feedback before adding a test + merging.

_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

Use `aria-haspopup="dialog"` and a stateful `aria-expanded` to provide useful info on the invoker button 

#### List the issues that this change affects.

Fixes https://github.com/github/primer/issues/3691

#### Risk Assessment

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **Fixes axe scan violation** - This change fixes an existing [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation.
- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.
- **New axe violation** - This change introduces a new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation. Please describe why the violation cannot be resolved below.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/alpha/select_panel_element.ts [hunk 1]
```diff
@@ -466,6 +466,7 @@ export class SelectPanelElement extends HTMLElement {
     if (event.target === this.dialog && event.type === 'close') {
       // Remove data-ready so it can be set the next time the panel is opened
       this.dialog.removeAttribute('data-ready')
+      this.invokerElement?.setAttribute('aria-expanded', 'false')
```
Comment (kendallgassner) [inline app/components/primer/alpha/select_panel_element.ts:469]: Can we move this to line `56` in the `close` eventListener?

Comment (JoyceZhu) [reply app/components/primer/alpha/select_panel_element.ts:469]: FYI, I just tried this out and for some reason the state doesn't correctly change back to being `aria-expanded=false` if I use Esc to close the menu

Comment (kendallgassner) [reply app/components/primer/alpha/select_panel_element.ts:469]: maybe we should put it in both places. I am thinking that if a consumer closes the panel outside of the component this call might be missed.

**Review by kendallgassner — COMMENTED** [review]
Approved! We can merge after testing 👍

**Review by kendallgassner — APPROVED** [review]
Whoops

**Pull request merged**